src/main/kotlin/app/revanced/patches/youtube/misc/integrations/patch/IntegrationsPatch.kt

```diff
@@ -30,49 +30,30 @@ class IntegrationsPatch : BytecodePatch(
         InitFingerprint, StandalonePlayerFingerprint, ServiceFingerprint
     )
 ) {
-    override fun execute(context: BytecodeContext): PatchResult {
-        if (context.findClass("Lapp/revanced/integrations/utils/ReVancedUtils") == null) return PatchResultError("Integrations have not been merged yet. This patch can not succeed without the integrations.")
-
-        val fingerprints = arrayOf(InitFingerprint, StandalonePlayerFingerprint, ServiceFingerprint)
-
-        fingerprints.forEach {
-            val result = it.result
+    companion object {
+        private const val INTEGRATIONS_DESCRIPTOR = "Lapp/revanced/integrations/utils/ReVancedUtils;"
+    }
 
-            if (it.result == null) {
-                return PatchResultError("${it.name} failed to resolve")
+    override fun execute(context: BytecodeContext): PatchResult {
+        if (context.findClass(INTEGRATIONS_DESCRIPTOR) == null)
+            return PatchResultError("Integrations have not been merged yet. This patch can not succeed without merging the integrations.")
+
+        arrayOf(InitFingerprint, StandalonePlayerFingerprint, ServiceFingerprint).map {
+            it to (it.result ?: return PatchResultError("${it.name} failed to resolve"))
+        }.forEach { (fingerprint, result) ->
+            with(result.mutableMethod) {
+                // parameter which holds the context
+                val contextParameter = if (fingerprint == ServiceFingerprint) parameters.size else 1
+                // register which holds the context
+                val contextRegister = implementation!!.registerCount - contextParameter
+
+                addInstruction(
+                    0,
+                    "sput-object v$contextRegister, $INTEGRATIONS_DESCRIPTOR->context:Landroid/content/Context;"
+                )
             }
-
-            val method = result!!.mutableMethod
-            val register = method.implementation!!.registerCount - if (it == ServiceFingerprint) {
-                method.parameters.size
-            } else 1
-
-            method.addInstruction(
-                0,
-                "sput-object v$register, Lapp/revanced/integrations/utils/ReVancedUtils;->context:Landroid/content/Context;"
-            )
         }
 
-
-        val classDef = InitFingerprint.result!!.mutableClass
-        classDef.methods.add(
```

Comment:
This function is used in integrations which should be removed.

Reply:
This has already been removed in commit https://github.com/revanced/revanced-patches/pull/828/commits/5d8e2f5a947911bf17f412ec7d0d4f1f71139153.

Reply:
https://github.com/revanced/revanced-integrations/blob/main/app/src/main/java/app/revanced/integrations/utils/ReVancedUtils.java#L64

Reply:
Should the integrations not use [ReVancedUtils.getAppContext()](https://github.com/revanced/revanced-integrations/blob/main/app/src/main/java/app/revanced/integrations/utils/ReVancedUtils.java#L20) instead?

Reply:
Both of functions are identical so yeah.

Reply:
Can you open a PR for that?